File: Lifecycle-aware/LifecycleAware1/app/src/main/java/jalilurrahman/com/lifecycleaware1/MainActivity.java
package jalilurrahman.com.lifecycleaware1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        getLifecycle().addObserver(ActivitylifecycleawareMain.getInstance());
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        getLifecycle().removeObserver(ActivitylifecycleawareMain.getInstance());
    }

    public void secondExample(View view){
        Intent intent = new Intent(this,SecondExample.class);
        startActivity(intent);
    }
}

File: Lifecycle-aware/LifecycleKotlin/app/src/main/java/jalilurrahman/com/lifecyclekotlin/MainActivity.kt
package jalilurrahman.com.lifecyclekotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import jalilurrahman.com.lifecyclekotlin.locationManager.LocationManagerActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun LocationManagerAction(view:View) {
        var intent = Intent(this@MainActivity,LocationManagerActivity::class.java)
        startActivity(intent)
    }
}
